src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs

```diff
@@ -442,4 +441,64 @@ private void OnViewControlState(object obj, View.ControlStateChangedEventArgs co
             }
         }
     }
+
+    /// <summary>
+    /// The selector item class that stores a state-value pair.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class SelectorItem<T>
+    {
+        /// <summary>
+        /// The default constructor.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public SelectorItem() {}
+
+        /// <summary>
+        /// The constructor with the specified state and value.
+        /// </summary>
+        /// <param name="state">The state</param>
+        /// <param name="value">The value associated with state.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public SelectorItem(ControlState state, T value)
+        {
+            State = state;
+            Value = value;
+        }
+
+        /// <summary>
+        /// The state
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ControlState State { get; set; }
```

Comment:
How about just providing getter. The setter seems not necessary.

Reply:
If setter is removed, you can't set `State` in xaml. parser will throw exception.

Reply:
I see :)
Thank you for the explanation.
I'll approve this code but please think about the issue below.

What I was worrying about is that user may change value in SelectorItem after it applied to the View.
```
var item = new SelectorItem<Color>(ControlState.Pressed, Color.Black);
var colorSelector = new Selector<Color>();
colorSelector.Add(item);

var style = new ViewStyle() { BackgroundColor = colorSelector };

....

item.Value = Color.Red;
```
It is not likely to happen, but I want to avoid these complicate situation as possible.
Maybe providing setter as an internal API is the solution but current xaml parser won't allow internal API of the class that is not a root element. (I think it is a bug)